trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf5340/config.cmake

```diff
@@ -9,3 +9,4 @@
 set(SECURE_UART1                        ON         CACHE BOOL      "Enable secure UART1")
 set(PSA_API_TEST_TARGET                 "nrf"      CACHE STRING    "PSA API test target")
 set(ITS_NUM_ASSETS                      "5"        CACHE STRING    "The maximum number of assets to be stored in the Internal Trusted Storage area")
+set(NRF_NS_STORAGE                      OFF        CACHE BOOL      "Enable non-secure storage partition")
```

Comment:
Does this enable the storage partition? Or does it make an existing storage partition available as non-secure?

Reply:
The upstream TF-M doesn't know the concept of the partition manager etc. It just defines fixed memory regions which are either marked as secure or non-secure. This option basically adds another such a region. The region is hard-code in the upstream to match the storage partition from the device tree, and it's replaced with the actual storage partition (defined with the partition manager) in the NCS.

Reply:
This looks as good as it can be.

What we really need is for upstream tf-m to have a way to define multiple non-secure flash partitions. Currently, TF-M assumes there is a single continuous flash area that is used by the non-secure image.

The solution proposed here is definitely not scalable. In future scenarios there might be even more areas that need to be set as non-secure. What we could possibly do is to use (system) device tree for this, and represent there all flash map configuration with (possibly) multiple secure or non-secure areas.
CC @mbolivar-nordic